extensions/native/guest-macro/src/parse_compiler_output.rs

```diff
@@ -0,0 +1,65 @@
+use openvm_instructions::instruction::Instruction;
+use openvm_native_serialization::deserialize_instructions;
+use p3_field::{Field, PrimeField32};
+
+use crate::parse_kernel::ParsedKernel;
+
+#[derive(Debug)]
+pub struct CompiledKernelArgument {
+    pub name: String,
+    pub rust_type: String,
+    pub edsl_type: String,
+    pub fp: usize,
+}
+
+#[derive(Debug)]
+pub struct CompiledKernel<F: Field> {
+    pub function_name: String,
+    pub arguments: Vec<CompiledKernelArgument>,
+    pub body: Vec<Instruction<F>>,
+    pub rust_return_type: String,
+    pub edsl_return_type: String,
+    pub return_fp: usize,
+}
+
+pub fn parse_compiled_kernel<F: PrimeField32>(
+    parsed_kernel: ParsedKernel,
+    compiler_output: String,
```

Comment:
already mentioned above, but it's a bit weird to go through String for `Vec<u32>`, if you just had `Vec<u8>`, you could directly be chunking `u32` or even using `read_exact`